maven/WFLY-14934_Server_Plugin.adoc

```diff
@@ -0,0 +1,208 @@
+an application image for any WildFly release (even for a locally built SNAPSHOT) without having to wait for the 
+WildFly builder image to be built/deployed. 
+
+In addition, the WildFly server located inside the builder image, enforces a set of configuration API (a set of environment variables)
+and bash launch scripts. At startup time, a complex CLI script generation that impacts the server startup time is mandatorily executed.
+Having this level of configurability should be a user choice and not be enforced by the WildFly s2i image.
+
+To solve these pain points, we are removing in WFLY-14936 the server installation and its configuration aspects from the image.
+
+The WildFly S2I image becomes a generic image allowing the installation and execution of any provisioned WildFly server.
+
+Due to this removal, we need a way for customers to provision the WildFly server to be installed inside the new image 
+when building their application. This WildFly server will run the built application inside the image.
+
+To do so we are defining Maven tooling that handles the Galleon provisioning of the server, the application deployment and the configuration aspects.
+
+== Requirements
+
+The existing https://github.com/wildfly/wildfly-maven-plugin/[WildFly Maven plugin] has been designed to interact (mainly) with a running server instance. 
+We are here proposing to evolve the capabilities of this plugin for one to be able to build a fully configured server containing application deployment.
+ 
+This proposal adds a new kind of installation that is operated and controlled by the pom.xml file.  The pom.xml file contains all the information needed
+to control the server and application content and versioning.
+
+=== Hard Requirements
+
+==== Plugin requirements
+
+* The Maven plugin produces inside the project target dir a directory (named ```server``` by default) that contains the server and any deployment.
+
+==== WildFly Maven Plugin new goals
+
+* ```wildfly:package```: Galleon provisioning of a server, CLI script execution, copy content to the server, deploy application (if present). 
+* ```wildfly:provision```: Galleon provisioning of a server.
+
+In addition to these 2 new goals, we are replacing the existing automatic download and unzip of wildfly-dist by a Galleon provisioning of 
+WildFly. This change impacts ```run``` and ```start```goals).
+ 
+==== Galleon related Plugin configuration items
+
+These configuration items are common to the ```provision``` and ```package``` goals. They are used to configure 
+Galleon to provision a server.
+
+* feature-packs: A list of feature-pack to install.
+* log-provisioning-time: Whether to log provisioning time at the end.
+* offline-provisioning: Whether to use offline mode when the plugin resolves an artifact.
+* galleon-options: Arbitrary Galleon options used when provisioning the server.
+* provisioning-dir: By default the server is generated in ```target/server``` directory. 
```

Comment:
Neat-picking but:

* provisioning-dir: the name of the directory relative to the Maven project build directory when the server is provisioned. It defaults to `server` and will result in the server being provisioned in (`${project.build.directory}/server` which is resolved to `target/server` by default)

Reply:
Thanks, done.